datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1562,6 +1584,20 @@ pub(crate) mod tests {
         .build_arc()
     }
 
+    pub(crate) fn parquet_exec_with_stats() -> Arc<ParquetExec> {
+        let mut statistics = Statistics::new_unknown(&schema());
+        statistics.num_rows = Precision::Inexact(10);
+        statistics.column_statistics = column_stats();
+
+        let config =
+            FileScanConfig::new(ObjectStoreUrl::parse("test:///").unwrap(), schema())
+                .with_file(PartitionedFile::new("x".to_string(), 10000))
+                .with_statistics(statistics);
+        assert_eq!(config.statistics.num_rows, Precision::Inexact(10));
+
+        ParquetExec::builder(config).build_arc()
+    }
```

Comment:
Note that we are now providing statistics in the parquet exec. This was preventing the coalesce insertion (as seen in our real world reproducer).